legend-engine-xt-changetoken-pure/src/main/resources/core_pure_changetoken/changetoken_test.pure

```diff
@@ -65,6 +115,257 @@ function meta::pure::changetoken::tests::getVersions():Versions[1]
+function meta::pure::changetoken::tests::getVersionsCustom():Versions[1]
+{
+  ^Versions(
+    versions=[
+      ^Version( // base version
+        version='ftdm:abcdefg123'
+      ),
+      ^Version(
+        version='ftdm:abcdefg456',
+        prevVersion='ftdm:abcdefg123',
+        changeTokens=[
+          ^AddField(
+            class='meta::pure::changetoken::tests::SampleClass',
+            fieldName='abc',
+            fieldType='Custom[1]',
+            safeCast=true,
+            defaultValue=^ConstValue(value='{"@type":"Custom","value":"0d"}')
+          )
+        ]
+      )
+    ]
+  );
+}
+
+// This is test Versions: AddField with ConstValue('{"@type":"Custom","restricted":true,"value":0,"range":{"@type":"intMinMax","min":-1,"max":1,"round":0.5}}')
+function meta::pure::changetoken::tests::getVersionsCustomNested():Versions[1]
+{
+    ^Versions(
+            versions=[
+                ^Version( // base version
+                        version='ftdm:abcdefg123'
+                ),
+                ^Version(
+                        version='ftdm:abcdefg456',
+                        prevVersion='ftdm:abcdefg123',
+                        changeTokens=[
+                            ^AddField(
+                                    class='meta::pure::changetoken::tests::SampleClass',
+                                    fieldName='abc',
+                                    fieldType='Custom[1]',
+                                    safeCast=true,
+                                    defaultValue=^ConstValue(value='{"@type":"Custom","restricted":true,"value":0,"range":{"@type":"intMinMax","min":-1,"max":1,"round":0.5}}')
+                            )
+                        ]
+                )
+            ]
+    );
+}
+
+// This is test Versions: AddField with ConstValue(^Map<String, Any>()->put('@type', 'Custom')->put('restricted', true)->put('value', 0)->put('range', ^Map<String, Any>()->put('@type', 'intMinMax')->put('min', -1)->put('max', 1)->put('round', 0.5)))
+function meta::pure::changetoken::tests::getVersionsCustomMap():Versions[1]
+{
+    ^Versions(
+            versions=[
+                ^Version( // base version
+                        version='ftdm:abcdefg123'
+                ),
+                ^Version(
+                        version='ftdm:abcdefg456',
+                        prevVersion='ftdm:abcdefg123',
+                        changeTokens=[
+                            ^AddField(
+                                    class='meta::pure::changetoken::tests::SampleClass',
+                                    fieldName='abc',
+                                    fieldType='Custom[1]',
+                                    safeCast=true,
+                                    defaultValue=^ConstValue(value=^Map<String, Any>()->put('@type', 'Custom')->put('restricted', true)->put('value', 0)->put('range', ^Map<String, Any>()->put('@type', 'intMinMax')->put('min', -1)->put('max', 1)->put('round', 0.5)))
```

Comment:
You should use the native function newMap to instantiate maps instead of ^Map. For example, as below (though noting that I may not have got the parentheses correct).
```suggestion
                                    defaultValue=^ConstValue(value=newMapMap([pair('@type', 'Custom'), pair('restricted', true), pair('value', 0), pair('range', newMap([pair('@type', 'intMinMax'), pair('min', -1), pair('max', 1), pair('round', 0.5)])])
```